metaserver/client.go

```diff
@@ -19,24 +21,154 @@ func NewClient(addr string, client *http.Client) *Client {
 type Client struct {
 	addr   string
 	client *http.Client
+	token  string
+}
+
+func (client *Client) AuthorizeRenter(privateKey *rsa.PrivateKey, renterID string) error {
+	authClient := authorization.NewClient(client.addr, client.client)
+	token, err := authClient.GetAuthToken(privateKey, "renter", renterID)
+	if err != nil {
+		return err
+	}
+	client.token = token
+	return nil
+}
+
+func (client *Client) AuthorizeProvider(privateKey *rsa.PrivateKey, providerID string) error {
+	authClient := authorization.NewClient(client.addr, client.client)
+	token, err := authClient.GetAuthToken(privateKey, "provider", providerID)
+	if err != nil {
+		return err
+	}
+	client.token = token
+	return nil
+}
+
+func (client *Client) RegisterProvider(info *core.ProviderInfo) error {
+	url := fmt.Sprintf("http://%s/providers", client.addr)
+	body, err := json.Marshal(info)
+	if err != nil {
+		return err
+	}
+	resp, err := client.client.Post(url, "application/json", bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusCreated {
+		var respMsg postProviderResp
```

Comment:
Recommendation: Return an type errorResp struct{Error: string} when something goes wrong on the server then extract this error decoding logic into a single helper function. See renter's client.go.

Reply:
👍